src/python/pants/backend/scala/dependency_inference/scala_parser_test.py

```diff
@@ -252,18 +264,26 @@ def this(bar: SomeTypeInSecondaryConstructor) {
             "org.pantsbuild.example.HasPrimaryConstructor": FrozenOrderedSet(
                 ["bar", "SomeTypeInSecondaryConstructor"]
             ),
+            "org.pantsbuild.example.ApplyQualifier": FrozenOrderedSet(
+                ["Integer", "a", ".toInt", "calc.calcFunc"]
+            ), 
             "org.pantsbuild.example": FrozenOrderedSet(
                 ["ABaseClass", "ATrait1", "ATrait2.Nested", "BaseWithConstructor"]
             ),
         }
     )
 
     assert set(analysis.fully_qualified_consumed_symbols()) == {
-        # Because they contain dots, and thus might be fully qualified. See #13545.
+        # Select with Apply as Qualifier only yield the name after the Apply
+        ".toInt",
```

Comment:
This seems like a bug... is it avoidable by recursing on a different portion of `node.qual`?

Reply:
it is actually not related to the qual, it existed even before my pr, that the behavior of `extractName` when the qual is not one of the cases in the match, but let me try and fix that in another commit.

Reply:
@stuhood I've pushed a new commit that fixes the leading comma.

https://github.com/pantsbuild/pants/pull/15565/commits/9f5782b3007468b09ac3228f3657b40886fa3f40